packages/iris-grid/src/IrisGrid.tsx

```diff
@@ -2504,6 +2508,20 @@ class IrisGrid extends Component<IrisGridProps, IrisGridState> {
     onError(error);
   }
 
+  onViewportUpdate(): void {
+    const { model } = this.props;
+    // pending and no timer already exists
+    if (model.hasPendingOperations && this.pendingOperationsTimeout === null) {
+      this.pendingOperationsTimeout = setTimeout(() => {
```

Comment:
You'll need to clear this timeout on unmount as well. Similar to my other suggestion, I'd suggest a name `viewportLoadingTimeout` instead.